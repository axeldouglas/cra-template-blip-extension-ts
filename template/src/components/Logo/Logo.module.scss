.App-logo {
    pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
    .App-logo {
        animation: Heartbit infinite 4s linear;
    }
}

@keyframes Heartbit {
    50% {
        -webkit-transform: scale(0.5);
        opacity: 0.3;
    }
    100% {
        -webkit-transform: scale(1);
        opacity: 1;
    }
}