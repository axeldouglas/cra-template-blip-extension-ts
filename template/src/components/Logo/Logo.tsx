import classNames from 'classnames/bind';

import styles from './Logo.module.scss';

import logo from 'assets/img/blip-balloon.svg';

type Size = 'small' | 'medium' | 'large';

interface LogoProps {
    size?: Size;
}

const sizes: Record<Size, string> = {
    small: 'h-32',
    medium: 'h-40',
    large: 'h-48'
};

const cx = classNames.bind(styles);

function Logo({ size = 'medium' }: LogoProps) {
    return (
        <img className={cx('App-logo', sizes[size])} src={logo} alt="BLiP" />
    );
}

export default Logo;
