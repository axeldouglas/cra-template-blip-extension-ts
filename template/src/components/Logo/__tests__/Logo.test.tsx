import { render, screen } from '@testing-library/react';

import Logo from '..';

describe('<Logo />', () => {
    it('should render the medium Logo as default', () => {
        const { container } = render(<Logo />);

        expect(screen.getByRole('img', { name: /blip/i })).toBeInTheDocument();

        expect(container.firstChild).toHaveClass('App-logo h-40');

        expect(container.firstChild).toMatchSnapshot();
    });

    it('should render the small Logo', () => {
        const { container } = render(<Logo size="small" />);

        expect(screen.getByRole('img', { name: /blip/i })).toBeInTheDocument();

        expect(container.firstChild).toHaveClass('App-logo h-32');

        expect(container.firstChild).toMatchSnapshot();
    });

    it('should render the medium Logo', () => {
        const { container } = render(<Logo size="medium" />);

        expect(screen.getByRole('img', { name: /blip/i })).toBeInTheDocument();

        expect(container.firstChild).toHaveClass('App-logo h-40');
    });

    it('should render the large Logo', () => {
        const { container } = render(<Logo size="large" />);

        expect(screen.getByRole('img', { name: /blip/i })).toBeInTheDocument();

        expect(container.firstChild).toHaveClass('App-logo h-48');

        expect(container.firstChild).toMatchSnapshot();
    });
});
