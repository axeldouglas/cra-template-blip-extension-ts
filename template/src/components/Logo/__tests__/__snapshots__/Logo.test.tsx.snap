// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render the large Logo 1`] = `
<img
  alt="BLiP"
  class="App-logo h-48"
  src="[object Object]"
/>
`;

exports[`<Logo /> should render the medium Logo as default 1`] = `
<img
  alt="BLiP"
  class="App-logo h-40"
  src="[object Object]"
/>
`;

exports[`<Logo /> should render the small Logo 1`] = `
<img
  alt="BLiP"
  class="App-logo h-32"
  src="[object Object]"
/>
`;
